code/modules/mob/living/living_defense.dm

```diff
@@ -377,6 +377,10 @@
 	else
 		stun_absorption[key] = list("end_time" = world.time + duration, "priority" = priority, "stuns_absorbed" = 0, \
 		"visible_message" = message, "self_message" = self_message, "examine_message" = examine_message)
+	addtimer(src, "remove_stun_absorption", duration)
+
+/mob/living/proc/remove_stun_absorption()
+	stun_absorption = null

```

Comment:
If you still have the examine text, you still have the absorption, because that ALSO checks.